packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,99 @@
+
+![Basic sign-in dialog](/img/authenticators/java-authenticators-signinform.png)
+
+### 2: Authenticate the user credentials
+
+After a user initiates the sign-in flow by entering their username and password and then clicking **Sign In**, add the following code:
+
+```java
+// create an IDX Authentication Wrapper (can be an application scoped singleton)
+IDXAuthenticationWrapper idxAuthenticationWrapper = new IDXAuthenticationWrapper();
+
+// begin transaction
+// The proceedContext needs to be persisted between interactions (e.g. stored in a HttpSession for web apps)
+ProceedContext proceedContext = idxAuthenticationWrapper.begin().getProceedContext();
+// TODO: add device context
+
+// set the user's credentials
+AuthenticationOptions authenticationOptions = new AuthenticationOptions(username, password);
+
+// start authentication flow
+AuthenticationResponse authenticationResponse = idxAuthenticationWrapper.authenticate(authenticationOptions, proceedContext);
+```
+
+### 3: Handle the response from the sign-in flow
+
+The `AuthenticationResponse` contains the current state of the authentication process, it could be a `SUCCESS` or may indicate more information is needed to complete the flow, for example, require an additional factor.
+
+Query the `authenticationStatus` property of the `AuthenticationResponse` object returned by `authenticate` to discover the current status of the authentication process.
+
+```java
+// Update the context each time a request is made to Okta
+// make sure it is persisted for web applications
+proceedContext = authenticationResponse.getProceedContext();
+
+// get the authenticationStatus
+AuthenticationStatus authenticationStatus = authenticationResponse.getAuthenticationStatus();
+
+switch (authenticationStatus) {
+    case SUCCESS:
+        … your code …
+    case PASSWORD_EXPIRED:
+        … your code …
+}
+```
+
+### 4: Select a possible authenticator factor
+
+If you configured your Okta org correctly, you need to respond to two specific authenticator statuses to handle this scenario in addition to `SUCCESS` and `PASSWORD_EXPIRED`:
+
+* `AWAITING_PROFILE_ENROLLMENT` that is covered in this section
+* `AWAITING_AUTHENTICATOR_ENROLLMENT_SELECTION` that is covered in [the challenge flow section](#integrate-sdk-for-authenticator-challenge).
+
+The names of the authenticators available for enrollment or challenge can be found in the `AuthenticationResponse` object's `authenticators` collection. Display a list of authenticators to select the Google Authenticator for enrollment.
+
+> **Note:** The `isChallengeFlow` session variable is set to `false` if the user needs to enroll Google Authenticator first, and `true` if they have already done so.
+
+```java
+case AWAITING_AUTHENTICATOR_ENROLLMENT_SELECTION:
+
+    // There are multiple potential factors, for this example we are going to
+    // assume Google Authenticator (OTP) is used
```

Comment:
A couple of things. First, the use of "we" is ambiguous: the "royal" we, you the developer, you the developer and this bit of code, ... It doesn't really help the developer and doesn't really introduce a "conversational tone."

Next, if memory servers, the best way to determine if it's Google Authenticator is to use the ID property (I think that's the correct property) as the value is unambiguous: `"google_otp"`.

I'd suggest rewriting the code to use the named string rather than just "otp" as that applies to several authenticators. That may obviate the need for the comment. If not, or if there's no time for a rewrite (or if you'll leave that as an exercise for the reader), then please update the comment as it doesn't start with the subject:

// This example assumes that Google Authenticator is the only OTP authenticator in Okta org policy.

Reply:
```suggestion
    // There are multiple potential factors, for this it is assumed that
    // Google Authenticator (OTP) is used
```